### PR TITLE
add GL self-hosted example

### PR DESCRIPTION
**Pull request by casperdcl**
- add gitlab self-hosted example
- minor consistency (using `cml publish`)

### CODE REVIEW

File: content/docs/self-hosted-runners.md [hunk 1]
```diff
@@ -73,10 +76,48 @@ jobs:
...
+deploy-runner:
+  image: iterativeai/cml:0-dvc2-base1
+  script:
+    - |
+      cml runner \
+          --cloud=aws \
+          --cloud-region=us-west \
+          --cloud-type=p2.xlarge \
+          --cloud-spot \
+          --labels=cml-gpu
```
Comment (0x2b3bfa0) [inline content/docs/self-hosted-runners.md:98]: Not a fan of this barely idiomatic line continuation syntax, but... 🤷🏼‍♂️

**Review by 0x2b3bfa0 — APPROVED** [review]

**Comment by 0x2b3bfa0** [comment]
Sorry for the late review. Another pull request that flew under my radar. 🤦🏼‍♂️

File: content/docs/self-hosted-runners.md (hunk 1 above)
Comment (casperdcl) [reply content/docs/self-hosted-runners.md:98]: if you can configure the linter to accept any other reasonable format please do so :)

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:98]: `prettier --prose-wrap=preserve` or something similar? 🤔

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:98]: Still, it's a bit too broad.

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:98]: Cross–linking to https://github.com/iterative/cml/pull/779#issuecomment-952140244 for future references.

**Pull request merged**